subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java

```diff
@@ -100,24 +103,50 @@ public void visitFile(FileVisitDetails fileDetails) {
 
     private static void collectValidationProblems(Class<?> topLevelBean, Map<String, Boolean> problems, boolean enableStricterValidation) {
         boolean cacheable;
-        boolean mapErrorsToWarnings;
+        boolean canBeMadeCacheable;
+        boolean treatWarningsAsErrors;
         DocumentationRegistry documentationRegistry = new DocumentationRegistry();
         if (Task.class.isAssignableFrom(topLevelBean)) {
+            canBeMadeCacheable = true;
```

Comment:
How about `cachingSupported`?

Reply:
Also, instead of setting `canBeMadeCacheable`, why not just call `validateCacheableOrDisableCachingByDefaultReason()` here? You could even pass the cacheability annotation from here instead of reverse-engineering it from whether the type is a task or not in the method.

Reply:
The problem here is that we create the validation context afterwards, and I need the validation context for running the validation. I'll see what I can do.

Reply:
I moved the things around, now it should be easier to understand.